### DISCUSSION TITLE
SuperSet as a server for static docs for warehouse documentation

### DISCUSSION THREAD

**Discussion by andrewcstewart** [apache/superset#18496] [Ideas]
**Is your feature request related to a problem? Please describe.**

Not really a problem, more of an opportunity.  Users of BI tools can often benefit from access to documentation of how a particular data warehouse is organized (structure, field descriptions, etc).  Hosting this type of documentation for accessibility by BI users is not necessarily a challenging topic, but it is typically an under addressed.

A popular example of data source documentation is the auto-generated docs from [dbt](https://docs.getdbt.com/docs/dbt-cloud/using-dbt-cloud/cloud-generating-documentation).  Other examples might include simple Markdown documents rendered into html.  These sorts of static sites are simple enough to serve from S3, github pages, netlify, etc, but the complexity for most small data teams tends to come from adding an authentication layer.

**Describe the solution you'd like**

This idea may or may not be outside the scope of what SuperSet seeks to be, but it occurs to me that SuperSet could potentially be a natural platform to host such static documentation alongside charts, dashboards, SQL Lab, etc.  

One can imagine the SuperSet configuration including a setting for a directory where any html documents placed within will appear under a new section on the UI.  

**Describe alternatives you've considered**

Obviously static docs can be hosted via multiple different solutions.  The opportunity for co-locating warehouse documentation with the BI tools that rely upon such documentation is simply one of convenience.

**Comment by rumbin** [comment]
Interesting consideration.
More generally, I would wish for a closer integration of dbt's models into Superset. A static documentation is a good start, but having descriptions, meta information etc available even in Explore or SQLLab would be most beneficial, in my eyes.
Synchronization of dbt models and Superset' s datasets would be great.

Not trying to steal your thread, just adding some thoughts regarding dbt...

**Comment by arita37** [comment]
Can be easy to setup and very useful for numerous use cases

**Comment by zhaoyongjie** [comment]
Hi @andreamelloncelli, 

This is `Data Discovery`(also called `Data Dictionary`) area. But the Superset main goal is to provide a BI's solution. Please refer to this link for some solutions: https://medium.com/@torokyle/data-discovery-in-2020-3c907383caa0

**Comment by mistercrunch** [comment]
My take on this is that we should *integrate* with data catalog / data dictionaries / metadata graphs bidirectionally, through APIs and tooling.

I also really like the idea of making it easy to sync DBT models into Superset datasets. This should be pretty easy today with our APIs, it's a matter of reading DBT's `manifest.json` file and upserting datasets through the Superset API. That integration can live on a software package of its own, perhaps as a DBT extension, or a standalone thing.

------

I don't think it'd be reasonable to expand the scope of Superset to take on the metadata graph space. Sure we should have a nice global search feature that helps users find Superset objects, but I don't think it makes sense to go after cataloging everything.
